tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -32,105 +32,133 @@ var _ = Describe("odo devfile push command tests", func() {
-			}
-			Expect(valueFound).To(BeTrue())
+				Expect(valueFound).To(BeTrue())
+			})
+
+			It("update devfile and push again", func() {
+				// update devfile and push again
+				helper.ReplaceString("devfile.yaml", "name: FOO", "name: BAR")
+				helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass()
+			})
 
-			// update devfile and push again
-			helper.ReplaceString("devfile.yaml", "name: FOO", "name: BAR")
-			helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass()
 		})
 
 		When("JSON output is requested", func() {
-			It("should display output in JSON format", func() {
-
-				helper.Cmd("odo", "create", "nodejs", "--project", commonVar.Project, cmpName).ShouldPass()
 
-				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+			BeforeEach(func() {
 				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+				output = helper.Cmd("odo", "push", "-o", "json", "--project", commonVar.Project).ShouldPass().Out()
+			})
 
-				output := helper.Cmd("odo", "push", "-o", "json", "--project", commonVar.Project).ShouldPass().Out()
-				utils.AnalyzePushConsoleOutput(output)
+			It("should display push output in JSON format", func() {
 
-				// update devfile and push again
-				helper.ReplaceString("devfile.yaml", "name: FOO", "name: BAR")
-				output = helper.Cmd("odo", "push", "-o", "json", "--project", commonVar.Project).ShouldPass().Out()
 				utils.AnalyzePushConsoleOutput(output)
-
 			})
-		})
+			When("update devfile and push again", func() {
 
-		When("pushing devfile without an .odo folder", func() {
+				BeforeEach(func() {
+					helper.ReplaceString("devfile.yaml", "name: FOO", "name: BAR")
+					output = helper.Cmd("odo", "push", "-o", "json", "--project", commonVar.Project).ShouldPass().Out()
+				})
 
-			It("should be able to push based on name passed", func() {
-				helper.CopyExample(filepath.Join("source", "devfiles", "springboot", "project"), commonVar.Context)
-				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "springboot", "devfile.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
-				output := helper.Cmd("odo", "push", "--project", commonVar.Project, "springboot").ShouldPass().Out()
-				Expect(output).To(ContainSubstring("Executing devfile commands for component springboot"))
-			})
+				It("should display push updated output in JSON format", func() {
 
-			It("should error out on devfile flag", func() {
-				helper.Cmd("odo", "push", "--project", commonVar.Project, "--devfile", "invalid.yaml").ShouldFail()
+					utils.AnalyzePushConsoleOutput(output)
+				})
 			})
-
 		})
 
-		When("not in context directory", func() {
-			It("should push correct component based on --context flag", func() {
-				newContext := helper.CreateNewContext()
-				defer helper.DeleteDir(newContext)
+		When("not in context directory and doing odo push", func() {
```

Comment:
When("running odo push outside the context directory"

Reply:
updated